crypto/evp/signature.c

```diff
@@ -48,8 +48,6 @@ static void *evp_signature_from_algorithm(int name_id,
     int ctxfncnt = 0;
     /* Counts all init functions  */
     int initfncnt = 0;
-    /* Counts all performance functions (oneshot / update / final) */
-    int fncnt = 0;
```

Comment:
???

No objection to removing these but why did they get in in the first place?
I.e. was there a use for this which we might be losing?

Reply:
We have these counters in other `_from_algorithm` functions, and counters are a fairly simple way to check that enough has been implemented.  The `EVP_SIGNATURE` interface is overwhelmingly complex in comparison, and to my judgement, counters aren't enough to express the possible combinations in a readable manner, except for the simple cases (like, you must implement at least one init function, and the ctx functions are also easy to count.

Reply:
I don't think I understand but I'm willing to pass on this one.

Essentially, I've no idea if we've lost anything and I trust you to do the right thing.